src/js/components/Tabs/Tabs.js

```diff
@@ -29,16 +42,226 @@ const Tabs = forwardRef(
     const [activeIndex, setActiveIndex] = useState(rest.activeIndex || 0);
     const [activeContent, setActiveContent] = useState();
     const [activeTitle, setActiveTitle] = useState();
+    const [disableLeftArrow, setDisableLeftArrow] = useState();
+    const [disableRightArrow, setDisableRightArrow] = useState();
+    const [overflow, setOverflow] = useState();
+    const [focusIndex, setFocusIndex] = useState(-1);
+    const headerRef = useRef();
+    const size = useContext(ResponsiveContext);
 
     if (activeIndex !== propsActiveIndex && propsActiveIndex !== undefined) {
       setActiveIndex(propsActiveIndex);
     }
 
+    // Safari v15.5 has an issue with scrolling when overflow='hidden'
+    // and scroll-behavior='smooth'. For now we are detecting if the browser
+    // is safari to workaround this issue. The issue should be resolved soon
+    // and we can remove this. https://github.com/WebKit/WebKit/pull/1387
+    const isSafari = /^((?!chrome|android).)*safari/i.test(
+      window.navigator.userAgent,
+    );
+
     /* eslint-disable no-param-reassign */
     delete rest.activeIndex;
     delete rest.onActive;
     /* eslint-enable no-param-reassign */
 
+    const tabRefs = useMemo(
+      () => React.Children.map(children, () => React.createRef()),
+      [children],
+    );
+
+    // check if tab is in view
+    const isVisible = useCallback(
+      (element) => {
```

Comment:
I'm wondering if the code where we call this would be a bit simpler if we passed `index` here, instead of `element`. It would align more with `scrollToIndex` as well.

Reply:
Change to pass index in latest commit